### PR TITLE
Add birthtime_ns and nr of blocks and blocksize to btrfs stat output

### PR DESCRIPTION
**Pull request by Miauwkeru**

### CODE REVIEW

File: dissect/target/filesystems/btrfs.py [hunk 1]
```diff
@@ -173,8 +177,13 @@ def lstat(self) -> fsutil.stat_result:
...
         st_info.st_mtime_ns = entry.mtime_ns
         st_info.st_ctime_ns = entry.ctime_ns
+        st_info.st_birthtime_ns = entry.otime_ns
 
         # Btrfs has a birth time, called otime
         st_info.st_birthtime = entry.otime.timestamp()
 
+        # Add block information of the filesystem
+        st_info.st_blksize = self.fs.sector_size
+        st_info.st_blocks = self.fs.blocks
```
Comment (Schamper) [inline dissect/target/filesystems/btrfs.py:187]: `st_blocks` are the blocks for that file, not the entire filesystem.

**Review by Schamper — CHANGES_REQUESTED** [review]

File: dissect/target/filesystems/btrfs.py [hunk 2]
```diff
@@ -173,8 +174,13 @@ def lstat(self) -> fsutil.stat_result:
         st_info.st_atime_ns = entry.atime_ns
         st_info.st_mtime_ns = entry.mtime_ns
         st_info.st_ctime_ns = entry.ctime_ns
+        st_info.st_birthtime_ns = entry.otime_ns
```
Comment (Schamper) [inline dissect/target/filesystems/btrfs.py:177]: Can you put this below line 180.

File: tests/filesystems/test_btrfs.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+)
+def test_stat_information_file_blocksize(sector_size: int, filesize: int, expected_blocks: int) -> None:
+    entry = INode(Mock(), None)
+    entry.btrfs = Mock(sector_size=sector_size)
+    timestamp = c_btrfs.btrfs_timespec()
+    entry.inode = c_btrfs.btrfs_inode_item(
+        size=filesize, atime=timestamp, ctime=timestamp, otime=timestamp, mtime=timestamp
+    )
+
+    fs_entry = BtrfsFilesystemEntry(Mock(alt_seperator="/"), "unknown_path", entry)
```
Comment (Schamper) [inline tests/filesystems/test_btrfs.py:27]:
```suggestion
    fs_entry = BtrfsFilesystemEntry(Mock(), "some/path", entry)
```
There was a typo anyway (`alt_separator`) and it didn't error so I assume it's not needed anyway.

File: tests/filesystems/test_btrfs.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+@pytest.mark.parametrize(
+    "sector_size, filesize, expected_blocks",
+    [
+        (0x1000, 0x343, 0x1),
+        (0x1000, 0x1000, 0x1),
+        (0x1000, 0x1001, 0x2),
+    ],
+    ids=["lower", "equal", "greater"],
+)
+def test_stat_information_file_blocksize(sector_size: int, filesize: int, expected_blocks: int) -> None:
```
Comment (Schamper) [inline tests/filesystems/test_btrfs.py:19]: Might as well extend the test to test all stat fields?

**Review by Schamper — CHANGES_REQUESTED** [review]

File: tests/filesystems/test_btrfs.py (hunk 3 above)
Comment (pyrco) [reply tests/filesystems/test_btrfs.py]: It's not needed. It is currently only set in case of windows like filesystems

**Review by pyrco — APPROVED** [review]

**Pull request merged**